### DISCUSSION TITLE
JsonMapper's deserializer not removing properties

### DISCUSSION THREAD

**Discussion by LefanTan** [TypedProject/ts-express-decorators#2201] [Q&A] [answered]
Hi folks,

I'm trying to deserialize a TsED model (`RecordingModel`) into its prisma/client (`Recording`) equivalent so that it can be passed into prisma's update(). 

```
    // `recording` is of type RecordingModel, but we need to convert it to type Recording
    const prismaRecording = deserialize<Recording>(recording);
    this.logger.info("prismaRecording", prismaRecording);

    const updated = await this.recordingService.update({
      where: { id },
      data: prismaRecording,
    });
```

However, the logger is telling me that the `prismaRecording` is still of type `RecordingModel` which is causing some error since it contained some properties that doesn't exist in the schema. How can I get around this? 

Log
```
[2022-12-29T03:35:57.591] [INFO ] [TSED] - prismaRecording RecordingModel {
  id: 'b5bc5ba7-3616-4908-a9db-2b17cc2bafee',
  title: 'Vento - 29 December 2022',
  encodingStatus: 'DONE',
  videoUrl: 'http://localhost:9199/vento-assets/b5bc5ba7-3616-4908-a9db-2b17cc2bafee.mp4',
  user: null,
  userId: 'xZRUmiTMD8ZUI1ihrPaTQvrfh1M2',
  createdAt: 2022-12-29T10:17:53.848Z,
  updatedAt: 2022-12-29T10:17:55.436Z
}
```

**Comment by Romakita** [comment, marked as answer]
Why don’t you use the RecordModel directly…

also you missed to give the type to deserialize recording object. So deserialize function has no effect. Try this:

deserialize<Recording>(recording, { type: Recording })

see you

**Reply by LefanTan** [reply]
Thanks for the reply. I couldn't use `RecordModel` directly because RecordingModel contains additional properties (from the one to many relationship) that doesn't exist in the schema. For example the `User` property being passed to prisma update will cause it to fail. Using deserialize() to strip off these additional properties would make life a lot easier 😄 . 

Lint error when passing RecordingModel to update()'s data property. 
```
(property) data: (Prisma.Without<Prisma.RecordingUpdateInput, Prisma.RecordingUncheckedUpdateInput> & Prisma.RecordingUncheckedUpdateInput) | (Prisma.Without<Prisma.RecordingUncheckedUpdateInput, Prisma.RecordingUpdateInput> & Prisma.RecordingUpdateInput)
The data needed to update a Recording.

Type 'RecordingModel' is not assignable to type '(Without<RecordingUpdateInput, RecordingUncheckedUpdateInput> & RecordingUncheckedUpdateInput) | (Without<...> & RecordingUpdateInput)'.
  Type 'RecordingModel' is not assignable to type 'Without<RecordingUncheckedUpdateInput, RecordingUpdateInput> & RecordingUpdateInput'.
```

RecordingModel.ts
```
export declare class RecordingModel implements Recording {
    id: string;
    title: string;
    encodingStatus: ENCODING_STATUS;
    videoUrl: string | null;
    user: UserModel | null;
    userId: string | null;
    createdAt: Date;
    updatedAt: Date;
}

```

prisma/client/index.d.ts
```
export type Recording = {
  id: string
  title: string
  encodingStatus: ENCODING_STATUS
  videoUrl: string | null
  userId: string | null
  createdAt: Date
  updatedAt: Date
}

```

And doing `deserialize(recording, { type: Recording })` doesn't seem to work as type is expecting an object/class, and `Recording` is a type.

**Reply by Romakita** [reply]
Ok I see. But deserialize isn’t magic. If you don’t give a model, it won’t do anything. The solution is to use Groups decorator on property that you don’t to have for your db.
```ts
deserialize(recording, { type: RecordingModel, groups: ["recording"] })
```

then add `@Groups("!recording")` on each prop that you don’t wan’t in the database (like user prop).

see you
